src/applications/travel-pay/components/DocumentDownload.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { apiRequest } from '@department-of-veterans-affairs/platform-utilities/api';
+import environment from '@department-of-veterans-affairs/platform-utilities/environment';
+
+const DocumentDownload = ({
+  claimId,
+  documentId,
+  filename,
+  mimetype,
+  text,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+
+  const downloadLinkHandler = async (e, docId) => {
+    setError('');
+    if (isLoading) {
+      return;
+    }
+    setIsLoading(true);
+    e.preventDefault();
+
+    try {
+      const response = await apiRequest(
+        `${
+          environment.API_URL
+        }/travel_pay/v0/claims/${claimId}/documents/${docId}`,
+      );
+
```

Comment:
Maybe important, maybe not. vets-api is passing the filename and mimetype back in the headers. Is there a benefit to doing this on frontend? Maybe less props to pass around if we pull that stuff from the response headers

Reply:
"This" referring to the usage of the filename and mimetype from the document list response being passed into the component instead of getting them from the headers on this response? No, I don't think so. I'm down to have less props flying around, so I'll update this.